optuna_dashboard/preferential/_study.py

```diff
@@ -265,7 +266,16 @@ def get_best_trials(study_id: int, storage: optuna.storages.BaseStorage) -> list
     ]
     preferences = get_preferences(study_id, storage)
     worse_numbers = {worse for _, worse in preferences}
-    return [copy.deepcopy(t) for t in ready_trials if t.number not in worse_numbers]
+    study_system_attrs = storage.get_study_system_attrs(study_id)
+    best_trials = []
+    for t in ready_trials:
+        if t.number in worse_numbers:
+            continue
+        skipped_key = _SYSTEM_ATTR_PREFIX_SKIP_TRIAL + str(t._trial_id)
+        if skipped_key in study_system_attrs:
+            continue
+        best_trials.append(copy.deepcopy(t))
```

Comment:
I've just noticed that we do not need to declare ready_trials.

```suggestion
    for t in storage.get_all_trials(study_id, deepcopy=False, states=(TrialState.COMPLETE, TrialState.RUNNING)):
        if not t.system_attrs.get(_SYSTEM_ATTR_COMPARISON_READY, False):
            continue
        if t.number in worse_numbers:
            continue
        skipped_key = _SYSTEM_ATTR_PREFIX_SKIP_TRIAL + str(t._trial_id)
        if skipped_key in study_system_attrs:
            continue
        best_trials.append(copy.deepcopy(t))
```

Reply:
Also perhaps, it might be clearer to define `def is_skipped_trial(trial_id: int) -> bool` in `optuna_dashboard/preferential/_system_attrs.py`.